src/datepicker/hijri/ngb-date-hijri.ts

```diff
@@ -0,0 +1,85 @@
+import {NgbDate} from '../ngb-date';
+
+/**
+ * Base class for all the Hijri calendars.
+ */
+export abstract class NgbDateHijri extends NgbDate {
```

Comment:
I think this class should be removed and functionality moved to the calendar directly or the appropriate service (see my overall comment)